sma/speedwire/speedwireinterface.cpp

```diff
@@ -31,115 +31,161 @@
 #include "speedwireinterface.h"
 #include "extern-plugininfo.h"
 
-SpeedwireInterface::SpeedwireInterface(bool multicast, quint32 sourceSerialNumber, QObject *parent) :
+#include <QNetworkInterface>
+
+SpeedwireInterface::SpeedwireInterface(quint32 sourceSerialNumber, QObject *parent) :
     QObject(parent),
-    m_multicast(multicast),
     m_sourceSerialNumber(sourceSerialNumber)
 {
-
-    m_socket = new QUdpSocket(this);
-    connect(m_socket, &QUdpSocket::readyRead, this, &SpeedwireInterface::readPendingDatagrams);
-    connect(m_socket, &QUdpSocket::stateChanged, this, &SpeedwireInterface::onSocketStateChanged);
-    connect(m_socket, SIGNAL(error(QAbstractSocket::SocketError)),this, SLOT(onSocketError(QAbstractSocket::SocketError)));
-}
-
-SpeedwireInterface::~SpeedwireInterface()
-{
-    deinitialize();
-}
-
-bool SpeedwireInterface::initialize(const QHostAddress &address)
-{
-    if (m_initialized && !m_multicast) {
-        qCWarning(dcSma()) << "Try to initialize an already initialized speed wire interface. Please deinitialize() before calling this method again.";
-        return false;
+    m_unicast = new QUdpSocket(this);
+    connect(m_unicast, &QUdpSocket::readyRead, this, [=](){
+        QByteArray datagram;
+        QHostAddress senderAddress;
+        quint16 senderPort;
+
+        while (m_unicast->hasPendingDatagrams()) {
+            datagram.resize(m_unicast->pendingDatagramSize());
+            m_unicast->readDatagram(datagram.data(), datagram.size(), &senderAddress, &senderPort);
+
+            qCDebug(dcSma()).noquote() << "SpeedwireInterface: Unicast socket received data from" << QString("%1:%2").arg(senderAddress.toString()).arg(senderPort);
+            qCDebug(dcSma()) << "SpeedwireInterface: " << datagram.toHex();
+            emit dataReceived(senderAddress, senderPort, datagram, false);
+        }
+    });
+
+    connect(m_unicast, &QUdpSocket::stateChanged, this, [=](QAbstractSocket::SocketState socketState){
+        qCDebug(dcSma()) << "SpeedwireInterface: Unicast socket state changed" << socketState;
+    });
+
+    connect(m_unicast, static_cast<void (QUdpSocket::*)( QAbstractSocket::SocketError )>(&QAbstractSocket::error), this, [=](QAbstractSocket::SocketError error){
+        qCWarning(dcSma()) << "SpeedwireInterface: Unicast socket error occurred" << error << m_unicast->errorString();
+    });
```

Comment:
As mentioned above... should this perhaps do something? what if the network cable is unplugged?

Reply:
well, on an unicast socket that would have no impact regarding the binding. but I would like to see the error if one occurres in the logs.